samples/measure_reductions.exs

```diff
@@ -0,0 +1,29 @@
+map_fun = fn i -> [i, i * i] end
+
+Benchee.run(
+  %{
+    "flat_map" => fn input ->
+      if rem(Enum.random(1..5), 2) == 0 do
```

Comment:
Is there a reason we have randomness in this benchmark? :thinking: I'd at least like to have a comment explaining it because otherwise it wouldn't normally pass as a good  benchmark imo :D

Reply:
I wanted to show the full statistics. Without randomness all measurements are the same.

Reply:
Makes sense I think a comment would still help a bit here though :)